v0/destinations/singular/util.js

```diff
@@ -168,6 +163,12 @@ const platformWisePayloadGenerator = (message, isSessionEvent) => {
     payload.c = "carrier";
   }
   payload = removeUndefinedAndNullValues(payload);
+
+  // by default, the value of openuri and install_source should be "", i.e empty string if nothing is passed
+  payload.openuri = message.properties.url || "";
+  if (platform === "android" || platform === "Android") {
+    payload.install_source = message.properties.referring_application || "";
+  }
```

Comment:
Is this common for both the session and nonSession events? Then we can put this in a commonplace.

Reply:
No. Both `openuri` and `install_source ` are exclusive to session events. Both properties essentially capture important data regarding a user's app session, i.e from what url what it triggered, and the source package installing the app.

Reply:
[Reference](https://support.singular.net/hc/en-us/articles/360048588672-Server-to-Server-S2S-API-Endpoint-Reference#Session_Notification_Endpoint)

Reply:
Then why you moved it to the section of nonSession events?

Reply:
Actually I have moved it to the very end as we usually strip empty values off the payload before sending but `openuri` and `install_source` were needed to be sent, whether or not a value was provided (an empty string "" is needed to be sent in case a value is not passed).

Reply:
addressed this. moved into session event specific context